surface/inventory/models.py

```diff
@@ -12,3 +13,72 @@ class Application(models.Model):
     director = models.ForeignKey('Person', blank=True, null=True, on_delete=models.SET_NULL, related_name='+')
     director_direct = models.ForeignKey('Person', blank=True, null=True, on_delete=models.SET_NULL, related_name='+')
     dev_lead = models.ForeignKey('Person', blank=True, null=True, on_delete=models.SET_NULL, related_name='+')
+
+
+class FindingInheritanceQS(models.QuerySet):
+    def get_children(self) -> list:
+        return [
+            getattr(m, m.content_source.model)
+            for m in self.prefetch_related("content_source__model__finding_ptr").select_related('content_source')
+        ]
+
+
+class Finding(models.Model):
```

Comment:
The idea behind the request is interesting and appealing, indeed. However, I think we should do it a bit differently. We should have another model like `StatusOverride`, containing the fields we need for our overrides, like `new_state`, `reason`, `author`, etc, and a FK to this `Finding`. Or a FK to the apps, depending on what is the reason behind your override. In your resyncs you would just look for an override and not update state.

The reason I'm trying to decouple the override from `Finding` is solely because the needs for these overrides will change a lot between environments and making a one-size-fits-all override will be impossible.

If accepted, this approach means we let each "implementor" decide what's best for them, in terms of overriding status. 

Thanks for the suggestion 🙏

Reply:
IMO, how to handle "syncs" is something very specific to each subclass of Finding, not to the parent model, so the override or whatever logic is used to decide whether or not status/severity sholud be updated would be handled in that subclass as well